core/src/main/java/com/datastax/oss/driver/internal/core/channel/ChannelFactory.java

```diff
@@ -236,7 +259,22 @@ protected void initChannel(Channel channel) {
               .sslHandlerFactory()
               .map(f -> f.newSslHandler(channel, address))
               .map(h -> pipeline.addLast("ssl", h));
+
+          NodeMetricUpdater nodeMetricUpdater;
+          if (node instanceof DefaultNode) {
```

Comment:
node could also be null (particularly in unit tests), `instanceof` check covers that for us.

Reply:
On a side note, if it's not null it's always an instance of `DefaultNode`. That is not meant to be customizable, the interface/implementation separation is just for hiding stuff from the public API.